src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvConnection.cs

```diff
@@ -135,21 +137,10 @@ private async Task StartCore()
                     // We're done with the socket now
                     _socket.Dispose();
 
-                    // Fire the connection closed token and wait for it to complete
-                    var waitForConnectionClosedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+                    // Ensure this always fires
+                    FireConnectionClosed();
```

Comment:
If we're going to continue firing ConnectionClosed for libuv output errors, we should do the same for the socket transport.

Reply:
What do you mean? Aren't we doing that? It always fires no?

Reply:
I'm now thinking about making this more like the socket transport instead. The important think is consistency.

Instead of calling FireConnectionClosed here, just don't dispose _socket here. Instead, close the uv_stream_handle only after both the output loop completed and

a) you've received a terminal read callback or
b) the connection was aborted